src/components/TxFlow/ModalProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import {
+  createContext,
+  type Dispatch,
+  type ReactElement,
+  type ReactNode,
+  type SetStateAction,
+  useState,
+  type ComponentProps,
+  useEffect,
+} from 'react'
+import ModalDialog from '@/components/common/ModalDialog'
+import TokenTransferFlow from '@/components/TxFlow/TokenTransfer/TokenTransferFlow'
+import RejectTx from '@/components/TxFlow/RejectTx'
+import NewTxMenu from '@/components/TxFlow/NewTx'
+import ReplaceTxMenu from '@/components/TxFlow/ReplaceTx'
```

Comment:
This part can be extracted into another module. Which could be imported and inserted here.